classes/notification/managerDelegate/EditorialReportNotificationManager.inc.php

```diff
@@ -47,19 +47,26 @@ public function __construct(int $notificationType) {
 	public function initialize(Context $context, DateTimeInterface $dateStart, DateTimeInterface $dateEnd) : void
 	{
 		$this->_context = $context;
+		$locale = $this->_context->getPrimaryLocale();
 		$dateStart = $dateStart;
 		$dateEnd = $dateEnd;
 
 		$dispatcher = Application::get()->getDispatcher();
 
-		$this->_editorialTrends = Services::get('editorialStats')->getOverview([
-			'contextIds' => [$this->_context->getId()],
-			'dateStart' => $dateStart->format('Y-m-d'),
-			'dateEnd' => $dateEnd->format('Y-m-d'),
-		]);
-		$this->_editorialTrendsTotal = Services::get('editorialStats')->getOverview([
-			'contextIds' => [$this->_context->getId()],
-		]);
+		$this->_editorialTrends = Services::get('editorialStats')->getOverview(
+			[
+				'contextIds' => [$this->_context->getId()],
+				'dateStart' => $dateStart->format('Y-m-d'),
+				'dateEnd' => $dateEnd->format('Y-m-d'),
+			],
+			$locale
```

Comment:
I'm wary of passing the `$locale` down in this way. It's ok here and there, but in cases like below with the stats, it becomes cumbersome as the call stack gets larger, and more and more arguments need to be added on in different places.

Instead, we should try to find a way to be able to specify a current working locale that would be used when `AppLocale::getLocale(...)` is called. Perhaps this could be invoked during the scheduled task or something?

@asmecher do you have any ideas about how we might be able to switch the current locale at runtime? This might be a necessary pre-requisite for some of the work on improving emails for 3.4/3.5 -- especially automated emails like this one which compile data to be sent out.

Reply:
Hi @NateWr, I also think it's better to avoid passing things down the road, I'm used to automatic DI (not forgetting that too much sugar might bring diabetes haha) =]
These non-variadic functions with many arguments and the ones with a generic `$args` receiving unknown values (such as the `getOverview` above), are things that I think could be refactored with a builder or a POPO/options class, mainly to keep things strongly-typed/safe... Would be great if PHP had a shorter way to create objects such as `new Options { contextId => 1 }`.

---

I thought about adding a `setLocale` to the `AppLocale` class, it would be easiest way to fix it, but I thought it would be too risky.
1. The PHP's `setlocale` isn't (very safe)[https://www.php.net/setlocale#:~:text=Warning].
2. Languages normally have a global locale, and also ways to specify a different one within their string/date/numeric libraries (which is somehow similar to this case)... And I think it's better to avoid what I wrote in my first comment, a `setlocale(localeThatINeed)` followed by a `setlocale(previousLocale)`, oh well, let's wait for our third opinion :D

Reply:
Oh wow, I think that `setlocale` thread-safety warning must've been added relatively recently -- that's bad!

We do currently use `setlocale`, and depend on it, and for what it's worth we haven't heard much about random-ish locale changes experienced by users. On one hand, I think that problem has to be solved at the programming language level, and on the other hand, I don't have a ton of faith that the PHP folks will be able to resolve it in the near future.

As for this particular issue, what about just moving the `__` call further up in the call stack? Have the `getWorkflowStageName` function return a locale key rather than a translated string, then have the calling code perform the translation, specifying the language if needed. I consider translations to be a close-to-the-UI operation, and it makes sense to me not to bury that act so deep in the business logic.

It's not a general solution and the problem is likely to crop up elsewhere, but it seems tidy in this case.

Reply:
I agree! It's better to avoid translating until the last possible moment. Just wanted to raise this issue to look for a standard =]

Reply:
Makes sense! In that case it is my fault for baking the translation into the `Stats::getOverview()` method at https://github.com/pkp/pkp-lib/pull/6087/files#diff-6e8f715bc9c5bd0df97e5edbd0be03d3R91.

Maybe the way to do it is to remove the `name` and use the `key` to find the appropriate locale string wherever that array is used.